### ISSUE TITLE
Implement reading the raster metadata from S3 URLs we get from the terracatalogueclient

### ISSUE THREAD

**Issue by JohanKJSchreurs** [VitoTAP/stac-catalog-builder#17]
To Be Decided: Implement reading the raster metadata from S3 URLs we get from the terracatalogueclient, if possible.

## Goal

We want to convert the data we get from OpenSearch to a STAC collection and STAC items but some parts of the metadata need to be read from the files.
If possible we would like to get the metadata from the rasters for products we query via terracatalogueclient.

##  Possible downside

Downloading/reading every file entirely is probably slow, but we may not need to read the raster for each individual product.

## Approach

One file, or a sample of a few files could be enough, unless the data really is different for each file and is really necessary to add.

If the data is always going to be the same for all raster files we only need to check one file. Tile sizes are likely to be such a case.

Maybe we can read only the metadata without downloading the entire file. Not sure if that is possible. To be confirmed.

Example that uses rasterio with the S3 URLs:
https://github.com/eea/clms-hrvpp-tools-python/blob/main/HRVPP_opensearch_demo/HRVPP%20catalogue%20and%20download%20demo.ipynb

## Documentation: How to download products

https://vitobelgium.github.io/terracatalogueclient/usage.html#download-products

**Comment by JeroenVerstraelen** [comment]
To be seen if this is required. We'd like to avoid this. 

Blocked by https://github.com/VitoTAP/stac-catalog-builder/issues/16. Once the stac-catalog-builder is ready we can find out if reading the metadata is required.

**Comment by JohanKJSchreurs** [comment]
For now we can postpone this, or even avoid it entirely.

Given that:

- we only need the tile size AKA raster shape (width & height) and the this is normally the same size for all tiles
- at present, we only need this for five collections from OpenSearch (OSCARS)

we can suffice with a simpler option.

For each collection, we can simply download a few GeoTIFF rasters via the WEkEO viewer (see link below). This viewer has a functionality to select and download rasters:
https://www.wekeo.eu/data

Mind you, we want to get complete tiles in order to check the tile size, so we need to select the rasters by tile ID instead of using the bounding box. (so we avoid tiles that are only partially within the bbox.)  The tiles seem to to be square so it is pretty easy to see whether they are complete or not.

Once we know the tile size, what we still need is a functionality to update the STAC items that we already have saved, either directly in the STAC-API or updating the static STAC files on disk and re-run the STAC upload process. The STAC upload works like an upsert statement so re-running it replaces the old content.
